cli-plugins/socket/socket.go

```diff
@@ -35,16 +39,40 @@ func randomID() string {
 	return hex.EncodeToString(b)
 }
 
-func accept(listener *net.UnixListener, conn **net.UnixConn) {
+// accept creates a new Unix socket connection
+// and sends it to the *net.UnixConn channel
+// it allows reconnects
+func accept(listener *net.UnixListener) <-chan *net.UnixConn {
+	connChan := make(chan *net.UnixConn, 1)
+
 	go func() {
-		for {
-			// ignore error here, if we failed to accept a connection,
-			// conn is nil and we fallback to previous behavior
-			*conn, _ = listener.AcceptUnix()
-			// perform any platform-specific actions on accept (e.g. unlink non-abstract sockets)
-			onAccept(*conn, listener)
+		const maxRetries = 10
+		const waitBetweenRetries = 100 * time.Millisecond
+
+		var conn *net.UnixConn
+		var err error
+
+		// retry accepting a connection if there was an error
+		for i := 0; i < maxRetries; i++ {
+			// this is a blocking call and will wait
+			// until a new connection is accepted
+			// or until the timout is reached
+			conn, err = listener.AcceptUnix()
+
+			if err != nil {
+				time.Sleep(waitBetweenRetries)
+				continue
+			}
+			break
 		}
+		// perform any platform-specific actions on accept (e.g. unlink non-abstract sockets)
+		onAccept(listener)
+		connChan <- conn
```

Comment:
However, these changes make it so that now, after a successful connection, we leave this loop and don't accept connections anymore.